### ISSUE TITLE
Downloading a preview file works from the second attempt

### ISSUE THREAD

**Issue by esraagamal6** [camunda/camunda#26313]
### Describe the bug
When attempting to download an image from a document preview, the download fails on the first attempt. Instead, an error message appears stating, "Unable to download document." However, the download succeeds if retried

### To Reproduce

1. Deploy a process with two userTasks first one has a form with a file picker and the second has a form with document preview
2. Start the process
3. Attach an image in the first userTask and complete the process
4. Open the second userTask and download the image

https://github.com/user-attachments/assets/1d84fdff-7427-4ead-978b-9d9ff0d3c006

### Actual behavior
"Unable to download document" error message is displayed and file is not downloaded, but can be downloaded from the second attempt

### Expected behavior
File can be downloaded from the first attempt

### Environment

- OS: [e.g. MacOS]
- Browser: [e.g. chrome, safari]
- Tasklist Version: SNAPSHOT

### Additional context

---------------------------------------------------------------------------------------------

### Definition of Ready - Checklist

- [ ] The bug has been reproduced by the assignee in the environment compatible with the provided one; otherwise, the issue is closed with a comment
- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] Necessary screenshots, screen recordings, or files are attached to the bug report

For UI changes required to solve the bug:

- [ ] Design input has been collected by the assignee

### After solving the issue - Answer and document in the comments the questions below

- [ ] What was the root cause for this bug?
- [ ] Is there anything we should do for the error to not happen anymore?

**Comment by marcosgvieira** [comment]
@esraagamal6 can you please inform if you are spinning up Tasklist together with Operate?

**Comment by esraagamal6** [comment]
> @esraagamal6 can you please inform if you are spinning up Tasklist together with Operate?

@marcosgvieira Yes this is a SaaS environment so by default operate is started. Should that be a problem?

**Comment by mo-okocha** [comment]
The issue was traced to the fact that the tasklist was using an in-memory datastore, which saves documents directly on the server it was created on. When the system tries to access the document from a different server (due to load balancing), it results in a 404 error because the document is not available there.

@sjiekak  will update the system to use GCP instead, so the documents will be stored centrally and accessible from any server.

**Comment by mo-okocha** [comment]
@esraagamal6 The fix is done, you can validate now

**Comment by esraagamal6** [comment]
Retested OK!
